src/js/components/pages/layout-text-editor.jsx

```diff
@@ -371,14 +377,16 @@ var SceneMonacoTextEditor = React.createClass({
         return compareNewPropsAndCurrentEditorCopy;
 
     },
-    componentWillReceiveProps: function (nextProps) {
-        /* HubSendActions.loadScene(nextProps._id);*/
-    },
+
     componentDidUpdate: function (previousProps, previousState) {
 
         try {
-            if (!_.isEqual(this.state.scene, previousState.scene))
+            var sceneChangeShouldUpdate = !_.isEqual(this.state.scene, previousState.scene);
+            var emptyEditorShouldUpdate = this.getMonacoEditorVersionOfScene() === null;
```

Comment:
Without this check here, this.refs.monaco.editor (.editor) can be null, leaving the editor never have the JSON doc added.